contracts/extensions/TokenSupplier.sol

```diff
@@ -0,0 +1,168 @@
+  /// @param _tokenSupplyCeiling Total amount of tokens to issue
+  /// @param _tokenIssuanceRate Number of tokens to issue per day
+  function initialise(uint256 _tokenSupplyCeiling, uint256 _tokenIssuanceRate) public isRoot {
+    require(lastPinged == 0, "token-supplier-already-initialised");
+
+    tokenSupplyCeiling = _tokenSupplyCeiling;
+    tokenIssuanceRate = _tokenIssuanceRate;
+    lastPinged = block.timestamp;
+    lastRateUpdate = block.timestamp;
+  }
+
+  /// @notice Update the tokenSupplyCeiling, cannot set below current tokenSupply
+  /// @param _tokenSupplyCeiling Total amount of tokens to issue
+  function setTokenSupplyCeiling(uint256 _tokenSupplyCeiling) public isRoot isContract {
+    try ColonyExtension(msg.sender).identifier() returns (bytes32 extensionId) {
+      IColonyNetwork colonyNetwork = IColonyNetwork(colony.getColonyNetwork());
+      address installation = colonyNetwork.getExtensionInstallation(extensionId, address(colony));
+      require(installation == msg.sender, "token-supplier-not-managed-extension");
+      require(extensionId == VOTING_HYBRID, "token-supplier-cannot-set-value");
+    } catch {
+      require(false, "token-supplier-no-identifier");
+    }
+
+    tokenSupplyCeiling = max(_tokenSupplyCeiling, tokenSupply);
+
+    emit TokenSupplyCeilingSet(tokenSupplyCeiling);
+  }
+
+  /// @notice Update the tokenIssuanceRate
+  /// @param _tokenIssuanceRate Number of tokens to issue per day
+  function setTokenIssuanceRate(uint256 _tokenIssuanceRate) public isRoot isContract {
+    try ColonyExtension(msg.sender).identifier() returns (bytes32 extensionId) {
+      IColonyNetwork colonyNetwork = IColonyNetwork(colony.getColonyNetwork());
+      address installation = colonyNetwork.getExtensionInstallation(extensionId, address(colony));
+      require(installation == msg.sender, "token-supplier-not-managed-extension");
+      require(
+        extensionId == VOTING_HYBRID || (
+          extensionId == VOTING_REPUTATION &&
+          block.timestamp - lastRateUpdate >= 4 weeks &&
+          _tokenIssuanceRate <= add(tokenIssuanceRate, tokenIssuanceRate / 10) &&
+          _tokenIssuanceRate >= sub(tokenIssuanceRate, tokenIssuanceRate / 10)
+        ),
+        "token-supplier-cannot-set-value"
+      );
+    } catch {
+      require(false, "token-supplier-no-identifier");
+    }
+
+    tokenIssuanceRate = _tokenIssuanceRate;
+    lastRateUpdate = block.timestamp;
+
+    emit TokenIssuanceRateSet(tokenIssuanceRate);
+  }
+
+  /// @notice Issue the appropriate amount of tokens
+  function issueTokens() public {
+    require(lastPinged > 0, "token-supplier-not-initialised");
+
+    address token = colony.getToken();
+    uint256 tokenSupply = ERC20Extended(token).totalSupply();
+
+    uint256 newSupply = min(
+      sub(tokenSupplyCeiling, tokenSupply),
+      wmul(tokenIssuanceRate, wdiv((block.timestamp - lastPinged), PERIOD))
```

Comment:
Remind me what terminology you would prefer?

Reply:
It's not the terminology, it's the 'every x hours, you can claim y tokens' rather than it being a continuous supply.

Reply:
Ok, I've made it `ISSUANCE_PERIOD` which may help? Not sure what to say except this is the period of time which defines the issuance rate.

Reply:
Defining the rate and how often it can pay out are two different things. If `issueTokens` is called and then called again 12 hours later, the transaction will fail (`token-supplier-nothing-to-issue`) but I am saying there should be half a day's worth of tokens for me to claim.

Reply:
Once again, I have been caught out by Wads.